pkg/tasker/tasker.go

```diff
@@ -0,0 +1,194 @@
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting ITellorFilterer instance")
+	}
+	sub, err := tellorFilterer.WatchNewChallenge(&bind.WatchOpts{}, sink, nil)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "getting NewChallenge channel")
+	}
+	return sink, sub, nil
+}
+
+func (mt *Tasker) sendWork(challenge *tellor.ITellorNewChallenge) {
+	if challenge.CurrentRequestId[0].Int64() > int64(100) || challenge.CurrentRequestId[0].Int64() == 0 {
+		level.Warn(mt.logger).Log("msg", "new current variables request ID not correct - contract about to be upgraded")
+		return
+	}
+	newChallenge := &mining.MiningChallenge{
+		Challenge:  challenge.CurrentChallenge[:],
+		Difficulty: challenge.Difficulty,
+		RequestIDs: challenge.CurrentRequestId,
+	}
+	for _, acc := range mt.accounts {
+		level.Info(mt.logger).Log("msg", "new challenge",
+			"addr", acc.Address.String(),
+			"challenge", fmt.Sprintf("%x", newChallenge.Challenge),
+			"difficulty", newChallenge.Difficulty,
+			"requestIDs", fmt.Sprintf("%+v", newChallenge.RequestIDs),
+		)
+		mt.workSinks[acc.Address.String()] <- &mining.Work{Challenge: newChallenge, PublicAddr: acc.Address.String(), Start: uint64(rand.Int63()), N: math.MaxInt64}
+	}
+
+}
+
+func (mt *Tasker) Start() error {
+	var err error
+	level.Info(mt.logger).Log("msg", "tasker has been started")
+	currentChallenge, err := mt.getCurrentChallenge()
+
+	if err != nil {
+		return errors.Wrap(err, "tasker getting the current challenge")
+	}
+
+	level.Info(mt.logger).Log("msg", "tasker is sending the initial challenge to the miner")
+	mt.sendWork(currentChallenge)
+
+	// Subscribe and wait until the context cancellation event.
+	var sink chan *tellor.ITellorNewChallenge
+	var sub event.Subscription
+
+	// Initial subscription.
+	for {
+		sink, sub, err = mt.getNewChallengeChannel()
+		if err != nil {
+			level.Error(mt.logger).Log("msg", "initial subscribing to NewChallenge events failed")
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		break
+	}
+
+	for {
+		select {
+		case <-mt.ctx.Done():
+			level.Info(mt.logger).Log("msg", "unsubscribing from NewChallenge events")
+			if sub != nil {
+				sub.Unsubscribe()
+			}
+			level.Info(mt.logger).Log("msg", "tasker shutdown complete")
+			return mt.ctx.Err()
+		case err := <-sub.Err():
+			if err != nil {
+				level.Error(mt.logger).Log(
+					"msg",
+					"new challenge subscription error",
+					"err", err)
+			}
+
+			// Trying to resubscribe until it succeeds.
+			for {
+				sink, sub, err = mt.getNewChallengeChannel()
+				if err != nil {
+					level.Error(mt.logger).Log("msg", "re-subscribing to NewChallenge events failed")
+					time.Sleep(1 * time.Second)
+					continue
+				}
+				break
+			}
+			level.Info(mt.logger).Log("msg", "re-subscribed to NewChallenge events")
+		case vLog := <-sink:
+			level.Info(mt.logger).Log("msg", "new challenge arrived, canceling current pending submissions")
```

Comment:
Actually lets remove this log as inside the submitter we will see another log if a submit was cancelled. Otherwise the logs shows regardless of whether there is a pending submit or not.
```suggestion
```